substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/phases/NativeImageInlineDuringParsingPlugin.java

```diff
@@ -96,29 +92,27 @@
 public class NativeImageInlineDuringParsingPlugin implements InlineInvokePlugin {
 
     public static class Options {
-        @Option(help = "Inline methods which folds to constant during parsing before the static analysis.") public static final HostedOptionKey<Boolean> InlineBeforeAnalysis = new HostedOptionKey<>(
-                false);
+        @Option(help = "Inline methods which folds to constant during parsing before the static analysis.")//
+        public static final HostedOptionKey<Boolean> InlineBeforeAnalysis = new HostedOptionKey<>(false);
 
     }
 
     private final boolean analysis;
     private final HostedProviders providers;
 
-    /* the map in which we place the result of a inlining decision */
-    private static final ConcurrentHashMap<AnalysisMethod, ConcurrentHashMap<Integer, InvocationResult>> dataInline = new ConcurrentHashMap<>();
+    /**
+     * The map in which we place the result of a inlining decision. During analysis we store
+     * information about method in separate map using {@link #callSitemap} and {@link #putIfAbsent}.
+     * After analysis we read information from this map to find inline decision we made using
+     * {@link #getResult}.
+     */
+    private static final ConcurrentHashMap<AnalysisMethod, ConcurrentHashMap<CallSite, InvocationResult>> dataInline = new ConcurrentHashMap<>();
 
```

Comment:
https://github.com/marjanasolajic/graal/pull/13#discussion_r482010185

I modified the data structure to collect directly CallSite objects with the invocation result. Also, methods for adding new results and searching for the result are modified.